libsolidity/formal/Predicate.cpp

```diff
@@ -321,6 +329,22 @@ optional<string> Predicate::expressionToString(smtutil::Expression const& _expr,
 			// Solver gave a concrete array but length is too large.
 		}
 	}
+	if (smt::isNonRecursiveStruct(*_type))
+	{
+		auto const& structType = dynamic_cast<StructType const&>(*_type);
+		solAssert(_expr.name == "tuple_constructor", "");
+		auto const& tupleSort = dynamic_cast<TupleSort const&>(*_expr.sort);
+		auto members = structType.structDefinition().members();
+		solAssert(tupleSort.components.size() == members.size(), "");
+		solAssert(_expr.arguments.size() == members.size(), "");
+		vector<string> elements;
+		for (unsigned i = 0; i < members.size(); ++i)
+		{
+			optional<string> elementStr = expressionToString(_expr.arguments.at(i), members[i]->type());
+			elements.push_back(members[i]->name() + ": " + (elementStr.has_value() ? elementStr.value() : ""));
```

Comment:
Is it OK to use empty string in the report when we could not translate the term?
Or should we use some special character, like `?` to indicate missing value?

Reply:
For arrays I used the name of the variable itself

Reply:
Not sure what you mean.
When you are dealing with variable of type array, but we cannot create the string representation (maybe it is too long), that this method returns an `optional<string>` with no value and then later the formatter deals with this.
However, if such array is a member of a struct, it would be nice to still report other members of the struct, so I need to assign something to the member (here empty string).

Alternatively, if we cannot report some member of a struct, we could give up on the whole struct as well.

Reply:
Ah, right, got it now.
I think in this case it's better to omit the : as well, otherwise it could potentially be confused with empty string if : "".
So:
{x: 0, y: true, z} would be an example

Reply:
OK, I have for example this: `s2 = {x: 23676, a}` for a struct with `uint x` and `uint[] a`.

Reply:
Ok nice!